### ISSUE TITLE
the get_list() method does not work if there is a configuration for Shadowsocks in inbounds

### ISSUE THREAD

**Issue by backtrac3r** [iwatkot/py3xui#25]
inbounds: List[py3xui.Inbound] = await api.inbound.get_list()
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.13/site-packages/py3xui/async_api/async_api_inbound.py", line 70, in get_list
    inbounds = [Inbound.model_validate(data) for data in inbounds_json]
                ~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "/opt/homebrew/lib/python3.13/site-packages/pydantic/main.py", line 596, in model_validate
    return cls.__pydantic_validator__.validate_python(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        obj, strict=strict, from_attributes=from_attributes, context=context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
pydantic_core._pydantic_core.ValidationError: 2 validation errors for Inbound
settings.clients.0.id
  Field required [type=missing, input_value={'method': '', 'password'...lvs8dk4990', 'reset': 0}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.9/v/missing
clientStats.0.id
  Input should be a valid string [type=string_type, input_value=6, input_type=int]
    For further information visit https://errors.pydantic.dev/2.9/v/string_type

**Comment by backtrac3r** [comment]
py3xui: 0.2.5
3x-ui: 2.4.5

**Comment by backtrac3r** [comment]
what should I do with this error?

**Comment by iwatkot** [comment]
@backtrac3r, this error means that some fields in the pydantic models are set incorrectly. And  it's just needed to change their typing.

On the [line 68](https://github.com/iwatkot/py3xui/blob/main/py3xui/api/api_inbound.py#L68), you have this:
```python
inbounds_json = response.json().get(ApiFields.OBJ)
```

Could you please share what's inside of the inbounds_json, so I can fix it?
You can just print it to the console and post it right here.

Thank you
Sincerely

**Comment by vladimirpichugin** [comment]
@iwatkot 
[{'id': 8, 'up': 0, 'down': 0, 'total': 0, 'remark': 'vless', 'enable': True, 'expiryTime': 0, 'clientStats': [{'id': 49, 'inboundId': 8, 'enable': True, 'email': '9wp79w8z', 'up': 0, 'down': 0, 'expiryTime': 0, 'total': 0, 'reset': 0}], 'listen': '', 'port': 45608, 'protocol': 'vless', 'settings': '{\n  "clients": [\n    {\n      "id": "fa2ee677-9bdf-4184-a3f0-2fc06e9c6fbb",\n      "flow": "",\n      "email": "9wp79w8z",\n      "limitIp": 0,\n      "totalGB": 0,\n      "expiryTime": 0,\n      "enable": true,\n      "tgId": "",\n      "subId": "8mppjc7vr17kxjqq",\n      "reset": 0\n    }\n  ],\n  "decryption": "none",\n  "fallbacks": []\n}', 'streamSettings': '{\n  "network": "tcp",\n  "security": "none",\n  "externalProxy": [],\n  "tcpSettings": {\n    "acceptProxyProtocol": false,\n    "header": {\n      "type": "none"\n    }\n  }\n}', 'tag': 'inbound-45608', 'sniffing': '{\n  "enabled": true,\n  "destOverride": [\n    "http",\n    "tls",\n    "quic",\n    "fakedns"\n  ],\n  "metadataOnly": false,\n  "routeOnly": false\n}'}, {'id': 9, 'up': 403326, 'down': 2015, 'total': 0, 'remark': 'vmess', 'enable': True, 'expiryTime': 0, 'clientStats': [{'id': 50, 'inboundId': 9, 'enable': True, 'email': 'rs677iym', 'up': 0, 'down': 0, 'expiryTime': 0, 'total': 0, 'reset': 0}], 'listen': '', 'port': 58324, 'protocol': 'vmess', 'settings': '{\n  "clients": [\n    {\n      "id": "72be2ef6-0fb6-40d1-85fa-5a93c2ba009c",\n      "email": "rs677iym",\n      "limitIp": 0,\n      "totalGB": 0,\n      "expiryTime": 0,\n      "enable": true,\n      "tgId": "",\n      "subId": "02cwmezdpyxflpdy",\n      "reset": 0\n    }\n  ]\n}', 'streamSettings': '{\n  "network": "tcp",\n  "security": "none",\n  "externalProxy": [],\n  "tcpSettings": {\n    "acceptProxyProtocol": false,\n    "header": {\n      "type": "none"\n    }\n  }\n}', 'tag': 'inbound-58324', 'sniffing': '{\n  "enabled": true,\n  "destOverride": [\n    "http",\n    "tls",\n    "quic",\n    "fakedns"\n  ],\n  "metadataOnly": false,\n  "routeOnly": false\n}'}, {'id': 10, 'up': 0, 'down': 0, 'total': 0, 'remark': 'trojan', 'enable': True, 'expiryTime': 0, 'clientStats': [{'id': 51, 'inboundId': 10, 'enable': True, 'email': 'laa59k9i', 'up': 0, 'down': 0, 'expiryTime': 0, 'total': 0, 'reset': 0}], 'listen': '', 'port': 55973, 'protocol': 'trojan', 'settings': '{\n  "clients": [\n    {\n      "email": "laa59k9i",\n      "enable": true,\n      "expiryTime": 0,\n      "flow": "",\n      "limitIp": 0,\n      "password": "gvlabXgHmC",\n      "reset": 0,\n      "subId": "jvji40kmjocm62dv",\n      "tgId": "",\n      "totalGB": 0\n    }\n  ],\n  "fallbacks": []\n}', 'streamSettings': '{\n  "network": "tcp",\n  "security": "none",\n  "externalProxy": [],\n  "tcpSettings": {\n    "acceptProxyProtocol": false,\n    "header": {\n      "type": "none"\n    }\n  }\n}', 'tag': 'inbound-55973', 'sniffing': '{\n  "enabled": true,\n  "destOverride": [\n    "http",\n    "tls",\n    "quic",\n    "fakedns"\n  ],\n  "metadataOnly": false,\n  "routeOnly": false\n}'}]

**Comment by iwatkot** [comment]
@vladimirpichugin @backtrac3r 

Could you please check the fix in the branch? Or you can manually change the typing for Client.id:

```python
id: int | str | None = Field(default=None)
```

It should work with your data, but be aware that you received the Client array from the API without their IDs, which is weird.

**Comment by sothug** [comment]
I caught a similar error, only with a different field (reset)
```
pydantic_core._pydantic_core.ValidationError: 2 validation errors for Inbound
settings.clients.1.reset
  Input should be a valid integer [type=int_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.9/v/int_type
settings.clients.2.reset
  Input should be a valid integer [type=int_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.9/v/int_type
```

Python 3.11.9
py3xui 0.2.6
3x-ui 2.4.5


I noticed that in py3xui, clients and their statistics are the same model, but there are different paths in json for getting the relevant information, which contain some common information and some information for which they are responsible. This causes a problem, to solve which you need to do one of two things:
- add the ability to be None for some fields
- create different models for clients and their statistics

**Comment by iwatkot** [comment]
@sothug, hey!

While developing the lib I created a separate class for ClientStat at first, but then I noticed that this is almost identical with the original Client class, so I merged them. So we can discuss separating them again if it's needed. 

Regarding the possible None parameters, they are very easy to change, so you can create a PR with the needed fields changed.

Sincerely

**Comment by sothug** [comment]
I looked at the Client and ClientTraffic models again and noticed that the field that caused the validation error was in both models.

The error started to occur after I manually changed the "Auto-Renew" value (reset field) in two clients in the panel and removed it. Removing the value from this field resulted in storing a Null value in the database, which can no longer be converted to the default value in py3xui. However, if you initially do not write anything in this (and possibly other) fields, then nothing will be stored in the database.

So, it seems to me that a more correct solution would be to separate the models as it is done in 3X-UI. However, a simpler solution would be to simply add the ability to set fields to None that can be deleted in the panel.
